src/MSIdentityScaffolding/Microsoft.DotNet.MSIdentity/MicrosoftIdentityPlatform/AppSettingsModifier.cs

```diff
@@ -104,194 +104,147 @@ public void ModifyAppSettings(ApplicationParameters applicationParameters, IEnum
-        {
-            if (!appSettings.TryGetValue("MicrosoftGraph", out var microsoftGraphToken))
+            if (appSettings.TryGetValue(key, out var apiToken))
             {
-                return DefaultApiBlock;
+                // block exists
+                var existingBlock = JObject.FromObject(apiToken);
+                return ModifyAppSettingsObject(existingBlock, inputParameters) ? existingBlock : null;
             }
 
-            var inputParameters = new Dictionary<string, string?>
-            {
-                { PropertyNames.Scopes, DefaultProperties.MicrosoftGraphScopes },
-                { PropertyNames.BaseUrl, DefaultProperties.MicrosoftGraphBaseUrl }
-            };
-
-            return ModifyAppSettingsToken(microsoftGraphToken, inputParameters) ? microsoftGraphToken : null;
+            // block does not exist, create a new one
+            return inputParameters;
         }
 
-        private JToken? GetModifiedDownstreamApiBlock(JObject appSettings)
+        private bool ModifyAppSettingsObject(JObject existingSettings, JObject inputProperties)
         {
-            if (!appSettings.TryGetValue("DownstreamApi", out var downstreamApiToken))
+            bool changesMade = false;
+            foreach ((var propertyName, var newValue) in inputProperties)
             {
-                return DefaultApiBlock;
+                changesMade |= UpdateIfNecessary(existingSettings, propertyName, newValue);
             }
 
-            var inputParameters = new Dictionary<string, string?>
-            {
-                { PropertyNames.Scopes, string.Empty }, // Only update if a value is not already present
-                { PropertyNames.BaseUrl, string.Empty }
-            };
-
-            return ModifyAppSettingsToken(downstreamApiToken, inputParameters) ? downstreamApiToken : null;
+            return changesMade;
         }
 
-        private JToken DefaultApiBlock => JToken.FromObject(new
-        {
-            BaseUrl = DefaultProperties.MicrosoftGraphBaseUrl,
-            Scopes = DefaultProperties.MicrosoftGraphScopes
-        });
-
         /// <summary>
         /// Updates property in appSettings block when either there is no existing property or the existing property does not match
         /// </summary>
-        /// <param name="token"></param>
+        /// <param name="block"></param>
         /// <param name="propertyName"></param>
         /// <param name="newValue"></param>
         /// <returns></returns>
-        private bool UpdateIfNecessary(JToken token, string propertyName, string? newValue)
+        internal static bool UpdateIfNecessary(JObject block, string propertyName, JToken? newValue)
         {
-            var existingValue = token[propertyName]?.ToString();
-            var update = GetUpdatedValue(propertyName, existingValue, newValue);
-            if (update != null)
+            var existingValue = block[propertyName];
```

Comment:
Does this throw if propertyName does not exist?

Reply:
Good catch